### ISSUE TITLE
Restoring unikernels

### ISSUE THREAD

**Issue by PizieDust** [robur-coop/mollymawk#36]
Currently a unikernel could either be running or destroyed.
We should have functionality that let's us save the state of a running unikernel, gracefully shut it down, and this state can be at some point restored in a new unikernel, presenting no noticeable differences between the new unikernel and the previous unikernel. It may be sufficient to understand what's most important to preserve about a unikernel before it is destroyed, persisting this information and later using it.

**Comment by hannesm** [comment]
This requires to store unikernel information into the mollymawk database. This is for sure useful in the future.

**Comment by PizieDust** [comment]
Yes, the idea I have in mind is when a user destroys, we should store arguments of the unikernel in a record under the users data. I assume the binary of the unikernel should still exist on the albatross server and perhaps the block file should also still exist somewhere on the albatross server machine. 
When a user performs a restoration, we should just launch the previous binary and previous block data with the stored arguments. 
Or, we can ask the user if they will like a new instance of the block device.

**Comment by hannesm** [comment]
The unikernel binary does not exist anymore on the server when the unikernel is not running.

Block devices persist - I suspect #32 will solve what we want :)

**Comment by hannesm** [comment]
And I guess since we now have a rollback we could extend this to restoring as well, using the same approach (and assume that the reproducible build database carries the unikernel build). This means we only have to store the uuid of the build (or the hash of the binary). Should be easy to do.

The underlying thing is: we require for full functionality to have the reproducible unikernel database online and mollymawk using it. But I think this simplifies mollymawk a lot and is a nice property anyways.

As suggested above, I'd store hashes of the unikernel binary instead of uuid of the builds. Somehow feels more appropriate (but we can do this later :))
